GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/NotifyUsersAndRetireInactiveAccountsJob.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+            DateTime threeYearsAgoMinusThirtyDays = threeYearsAgo.AddDays(30);
+            DateTime threeYearsAgoMinusSevenDays = threeYearsAgo.AddDays(7);
+
+            List<User> usersToSendReminders = dataRepository.GetAll<User>()
+                .Where(u => u.LoginDate == threeYearsAgoMinusThirtyDays || u.LoginDate == threeYearsAgoMinusSevenDays)
+                .ToList();
+
+            foreach (User user in usersToSendReminders)
+            {
+                // TODO: Update loginUrl
+                string daysRemaining = user.LoginDate == threeYearsAgoMinusThirtyDays ? "30" : "7";
+                emailSendingService.SendAccountRetirementNotificationEmail(user.EmailAddress, daysRemaining, "example url");
+            }
+
+        }
+
+        private void RetireUsers()
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+
+            List<User> usersToRetire = dataRepository.GetAll<User>()
+                .Where(u => u.LoginDate < threeYearsAgo)
+                .Where(u => !u.HasBeenAnonymised)
+                .ToList();
+
+            foreach (User user in usersToRetire)
+            {
+                user.Firstname = $"User{user.UserId}";
+                user.Lastname = $"User{user.UserId}";
+                user.JobTitle = "Anonymised";
+                user.EmailAddress = "Anonymised";
+                user.ContactFirstName = "Anonymised";
+                user.ContactLastName = "Anonymised";
+                user.ContactJobTitle = "Anonymised";
+                user.ContactOrganisation = "Anonymised";
+                user.ContactEmailAddress = "Anonymised";
+                user.ContactPhoneNumber = "Anonymised";
+                user.Salt = "Anonymised";
+                user.PasswordHash = "Anonymised";
+
+                user.HasBeenAnonymised = true;
+                user.Status = UserStatuses.Retired;
+
+                var actionsToAnonymise = new List<AuditedAction>(new []
+                    {
+                        AuditedAction.UserChangeEmailAddress, 
+                        AuditedAction.UserChangeName, 
+                        AuditedAction.UserChangeJobTitle, 
+                        AuditedAction.UserChangePhoneNumber, 
+                        AuditedAction.PurgeRegistration, 
+                        AuditedAction.PurgeUser, 
+                        AuditedAction.RegistrationLog
+                    });
+
+                List<AuditLog> userAuditLogs = dataRepository.GetAll<AuditLog>()
+                    .Where(al => al.OriginalUserId == user.UserId || al.ImpersonatedUserId == user.UserId)
+                    .Where(al => actionsToAnonymise.Contains(al.Action))
+                    .ToList();
+
+                foreach (AuditLog auditLog in userAuditLogs)
+                {
+                    auditLog.DetailsString = "Anonymised";
+                }

```

Comment:
I'd pull this whole section (lines 94-113) into a separate method, called something like `AnonymiseAuditLogsForUser`